src/syntax/Parser.js

```diff
@@ -1254,7 +1254,7 @@ export class Parser {
 
     let result = this.parseStatementList_(!strictMode);
 
-    if (!strictMode && this.strictMode_ && params)
+    if (!strictMode && this.strictMode_)
       StrictParams.visit(params, this.errorReporter_);
```

Comment:
I don't understand what `visit(null, this.errorReporter_);`  accomplishes here in the case that `params` tree is not.